### PR TITLE
using PATH_MAX

### PR DESCRIPTION
**Pull request by SvenJo**
Signed-off-by: Sven Johannsen <s.johannsen@bretgeld-engineering.de>

### CODE REVIEW

File: filepath.cc [hunk 1]
```diff
@@ -5,15 +5,15 @@
...
-  std::string buf(8192, '\0');
+  std::string buf(PATH_MAX, '\0');
   getcwd(buf.data(), buf.size());
   buf.resize(buf.find_first_of('\0'));
   return buf;
 }
 
 std::string get_realpath(std::string_view path) {
-  char buf[8192];
-  if (!realpath(std::string(path).c_str(), buf))
```
Comment (rui314) [inline filepath.cc:16]: It looks like the new code always returns a string of PATH_MAX length filled with NUL bytes. I think just replacing 8192 with PATH_MAX is better.

**Comment by sicherha** [comment]
Caution! `PATH_MAX` is a bit of a false friend; [relying on it is trickier than one might expect](https://eklitzke.org/path-max-is-tricky) as an actual path might be more than `PATH_MAX` characters long. `getcwd()` and `realpath()` in particular require some special handling. Let me see if I can whip up an alternative patch.

**Comment by Alcaro** [comment]
In POSIX 2008, realpath(path, NULL) calls malloc, so that one's easy.

But yes, getcwd is nontrivial. glibc getcwd has the same NULL-malloc feature, but that's nonstandard.

**Comment by sicherha** [comment]
I guess we can close this pull request, right?

**Pull request closed without merging**